diff --git a/FrontEnd/src/pages/Payouts/index.tsx b/FrontEnd/src/pages/Payouts/index.tsx
index 58fe897..5174848 100644
--- a/FrontEnd/src/pages/Payouts/index.tsx
+++ b/FrontEnd/src/pages/Payouts/index.tsx
@@ -76,6 +76,8 @@ function Index() {
             return acc;
         }, {});
 
+        console.log("some change");
+
         return (
             <CardBgWrapper number={`${dayNum}Day` as CardNumber}>
                 {/* <h2>{`${dayNum}-Day Payout Cycles`}</h2>
